### PR TITLE
[Site Design Revamp] Adds recommended designs section

### LINKED ISSUE #16393: [Site Design Revamp] Main View - Add recommended designs section
## Description

Add a larger horizontally scrolling section under the title with caption “Picked for you” to prominently present a list of recommended designs.

- If one of the six default verticals are chosen, the text under "Picked for you" will read "Best for [vertical name]", e.g. "Best for Food"
- If a default vertical isn't chosen, the text under "Picked for you" will read "Best for Blogging"
- The substituted word (Food, Blogging) should be capitalized

PRs should target the `feature/site-design-revamp` feature branch.

### PR DESCRIPTION
**Pull request by antonis**
Fixes #16393

## Description
This PR adds a larger horizontally scrolling section under the title with caption “Picked for you” to prominently present a list of recommended designs.

Note:
* Presenting actual recommendations based on verticals is pending to be implemented separately in `[Site Design Revamp] Determine recommended designs based on vertical` issue.
* Other alternative approaches were considered/tried. One involved [using a separate recycler on top but I've run into issues with the collapsing header implementation and decided to use just one recycler](https://github.com/wordpress-mobile/WordPress-Android/pull/16486#discussion_r867761709).

### Start the site creation flow and verify that the recommended section is shown

|Before (`feature/site-design-revamp`)|After|
|---|---|
|Screenshot_20220504_190123|Screenshot_20220509_115433|

### Verify that the UI of the page layout picker is unchanged

|Before (`feature/site-design-revamp`)|After|
|---|---|
|2Screenshot_20220504_120843|Screenshot_20220509_115516|

## Regression Notes
1. Potential unintended areas of impact
Page Layout Picker UI

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and the existing `ModalLayoutPickerViewModelTest`

3. What automated tests I added (or what prevented me from doing so)
Relied/modified on existing tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ovitrif — COMMENTED** [review]
LGTM so far 🎉 

Thank you @antonis for the progress you did on this 🙇

File: WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutCategoryAdapter.kt [hunk 1]
```diff
@@ -28,6 +30,7 @@ class LayoutCategoryAdapter(
     override fun getItemCount(): Int = items.size
 
     override fun onBindViewHolder(holder: LayoutsItemViewHolder, position: Int) {
+        holder.setIsRecyclable(false) // TODO: Find a better fix!
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutCategoryAdapter.kt]: I've added a suggestion branch based on this PR's branch: https://github.com/wordpress-mobile/WordPress-Android/tree/task/16393-recommended-designs--suggestion-get-item-view-type with this commit: https://github.com/wordpress-mobile/WordPress-Android/commit/df0d9ea9caa60ede657014b21ac6e8094c1d4221, which overrides the `getItemViewType` function to re-enable recycling the views.

**Review by mkevins — COMMENTED** [review]
This looks good and is working well for me so far in my testing. Also, the approach is reasonable, imo. I've added a suggestion for how to tackle the recycling issue for the different types of category rows.

Another alternative to consider would be to make the recommended row its own recycler view at the top level (as a preceding sibling outside of the `layoutsRecyclerView`).

File: WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutCategoryAdapter.kt (hunk 1 above)
Comment (antonis) [reply WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutCategoryAdapter.kt]: Thank you for looking into this and testing @mkevins 🙇 
Your suggestion looks good and works as expected. I've merged it on this PR. Thank you 🥇

Comment (antonis) [reply WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutCategoryAdapter.kt]: > Another alternative to consider would be to make the recommended row its own recycler view at the top level (as a preceding sibling outside of the layoutsRecyclerView).

I've started with the approach of using a separate recycler on top but I've run into issues with the collapsing header implementation and decided to use just one recycler 😅

**Review by ovitrif — COMMENTED** [review]
Code changes LGTM 🎉 !

Testing will follow 👍

**Comment by ovitrif** [comment]
> Testing will follow 👍

Works as expected! 🎉 

@antonis Should we :shipit: ? 🚀

**Comment by antonis** [comment]
Thank you for testing and reviewing the code @ovitrif 🙇 

> @antonis Should we :shipit: ? 🚀

I think we can merge on the feature branch with your approval 🚢

**Pull request merged**